src/unix/linux/process.rs

```diff
@@ -661,7 +662,7 @@ fn get_all_pid_entries(
     if !entry.is_dir() {
         return None;
     }
-    let pid = Pid::from(usize::from_str(&name.to_string_lossy()).ok()?);
+    let pid = Pid::from(usize::from_str(name.to_str()?).ok()?);
```

Comment:
For lossiness, the Unicode replacement character (�) is used, but that never results in a validly parsed `usize`, so might as well skip directly to returning an error, instead of allocating and then erroring anyway.

Reply:
True. Thanks for the explanations!